### PR TITLE
 EZP-30099: Define preferred user date format

### PR DESCRIPTION
**Pull request by adamwojs**
| Question      | Answer
| ------------- | ---
| Tickets       | [EZP-30099](https://jira.ez.no/browse/EZP-30099)
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | yes
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

User settings / datetime format.

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/lib/Form/Data/UserSettingUpdateData.php [hunk 1]
```diff
@@ -18,7 +18,6 @@ class UserSettingUpdateData
      * @var string
      */
     private $identifier;
-
```
Comment (mikadamczyk) [inline src/lib/Form/Data/UserSettingUpdateData.php]: This empty line is needed

File: src/bundle/DependencyInjection/Configuration/Parser/UserPreferences.php [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    {
+        $nodeBuilder
+            ->arrayNode('user_preferences')
+                ->info('User Preferences configuration.')
+                ->children()
+                    ->arrayNode('full_datetime_format')
+                        ->children()
+                            ->scalarNode('date_format')
+                                ->info('The date format')
+                                ->example('dd/MM/yyy')
```
Comment (webhdx) [inline src/bundle/DependencyInjection/Configuration/Parser/UserPreferences.php]: I suppose `yyy` will work but more valuable example would be `dd/MM/yyy`

File: src/bundle/DependencyInjection/Configuration/Parser/UserPreferences.php [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                                ->info('The time format')
+                                ->example('hh:mm')
+                            ->end()
+                        ->end()
+                    ->end()
+                    ->arrayNode('short_datetime_format')
+                        ->children()
+                            ->scalarNode('date_format')
+                                ->info('The date format')
+                                ->example('dd/MM/yyy')
```
Comment (webhdx) [inline src/bundle/DependencyInjection/Configuration/Parser/UserPreferences.php]: Same as above

File: src/bundle/DependencyInjection/Configuration/Parser/UserPreferences.php [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                        ->end()
+                    ->end()
+                    ->arrayNode('allowed_short_date_formats')
+                        ->info('List of allowed short date formats.')
+                        ->example(['label dd/MM/yyyy' => 'dd/MM/yyyy', 'label for mm/dd/yyyy' => 'mm/dd/yyyy '])
+                        ->prototype('scalar')->end()
+                    ->end()
+                    ->arrayNode('allowed_short_time_formats')
+                        ->info('List of allowed short time formats.')
+                        ->example(['label hh:mm' => 'hh:mm ', 'label for HH:mm' => 'HH:mm'])
```
Comment (webhdx) [inline src/bundle/DependencyInjection/Configuration/Parser/UserPreferences.php]: Unneeded space `'hh:mm '`

File: src/lib/UserSetting/Setting/DateTimeFormatSerializer.php [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    public function deserialize(string $value): ?DateTimeFormat
+    {
+        $value = json_decode($value, true);
+
+        if (!is_array($value)) {
+            return null;
+        }
+
+        return new DateTimeFormat(
+            isset($value['date_format']) ? $value['date_format'] : null,
```
Comment (mikadamczyk) [inline src/lib/UserSetting/Setting/DateTimeFormatSerializer.php]:
```suggestion
            $value['date_format'] ?? null,
```

File: src/lib/UserSetting/Setting/DateTimeFormatSerializer.php [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            isset($value['time_format']) ? $value['time_format'] : null
```
Comment (mikadamczyk) [inline src/lib/UserSetting/Setting/DateTimeFormatSerializer.php]:
```suggestion
           $value['time_format'] ?? null
```

**Pull request merged**